force-app/main/default/classes/Chainable.cls

```diff
@@ -87,4 +156,18 @@ public abstract class Chainable {
 			return originalContext;
 		}
 	}
+
+
+	public class DeferLimitException extends Exception {}
```

Comment:
Honestly I just though it would be easier to determine the reason of the issue by leveraging different exception types, if you think otherwise we can simplify by using a single generic exception like `DeferException`